code/Model/SiteTree.php

```diff
@@ -2711,6 +2712,43 @@ public function getPageLevel() {
 	}
 
 	/**
+	 * @param string $action
+	 * @return Controller
+	 */
+	public function getAssociatedController($action = null) {
```

Comment:
`getController` seems nice

If we could also `setController` that'd be even nicer to allow per instance overrides or something